modules/diskq/logqueue-disk-non-reliable.c

```diff
@@ -162,7 +163,19 @@ _move_messages_from_overflow(LogQueueDiskNonReliable *self)
         }
       else
         {
-          if (log_queue_disk_write_message(&self->super, msg))
+          ScratchBuffersMarker marker;
+          GString *serialized = scratch_buffers_alloc_and_mark(&marker);
```

Comment:
I think it would better, if you "allocate" the GString outside of the while loop and here just truncate the GString.
If can't put from overflow into qout, but can put into disk, then it needs "allocate" a GString.

Reply:
Thanks, fixed!